src/app/services/query.service.ts

```diff
@@ -445,72 +474,94 @@ export class QueryService {
 
       this.query = null;
       this.isPreparing_2 = false;
+      resolve(true);
     } else {
       window.setTimeout((function () {
-        this.prepareStep2();
+        this.prepare_2(resolve);
       }).bind(this), 500);
     }
   }
 
+  update_2b(): Promise<any> {
+    return new Promise((resolve, reject) => {
+      if (!this.isUpdating_1 && !this.isPreparing_2) {
+        this.isUpdating_2 = true;
+        // add time stamp to the queue,
+        // only when the time stamp is at the end of the queue, the count is updated
+        this.clearQueueOfCalls(this.queueOfCalls_2);
+        let timeStamp = new Date();
+        this.queueOfCalls_2.push(timeStamp.getMilliseconds());
+        // set flags to true indicating the counts are being loaded
+        this.isLoadingSubjectCount_2 = true;
+        this.isLoadingObservationCount_2 = true;
+
+        this.query = null; // clear query
+        // update the subject count and observation count in the 2nd step
+        const constraint_1_2: Constraint = this.constraintService.constraint_1_2();
+        this.resourceService.getCounts(constraint_1_2)
+          .subscribe(
+            (countResponse) => {
+              const index = this.queueOfCalls_2.indexOf(timeStamp.getMilliseconds());
```

Comment:
Is this queue of calls really needed?

Reply:
there is still some parallel calls, not every call is chained